dags/remote_automated_ampliseq.py

```diff
@@ -0,0 +1,373 @@
+            logging.info(f"Transferring compressed directory for study {tar_info['study']['study']}...")
+            study = tar_info["study"]
+            tar_filename = tar_info["tar_filename"]
+            local_tar_path = os.path.join(study['path'], tar_filename)
+            remote_tar_path = os.path.join(REMOTE_PATH, study['study'], tar_filename)
+            sftp_op = SFTPOperator(
+                task_id='transfer_compressed_directory_' + study['study'],
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SFTP,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                local_filepath=local_tar_path,
+                remote_filepath=remote_tar_path,
+                operation='put',
+                create_intermediate_dirs=True
+            )
+            sftp_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def extract_remote_directory(tar_info):
+            logging.info(f"Extracting remote directory for study {tar_info['study']['study']}...")
+            study = tar_info["study"]
+            tar_filename = tar_info["tar_filename"]
+            remote_tar_path = os.path.join(REMOTE_PATH, study['study'], tar_filename)
+            command = f'mkdir -p {REMOTE_PATH}/{study["study"]} && tar --strip-components=1 -xzf {remote_tar_path} -C {REMOTE_PATH}/{study["study"]} && rm {remote_tar_path}'
+            ssh_op = CustomSSHOperator(
+                task_id='extract_remote_directory_' + study['study'],
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SSH,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                command=command
+            )
+            ssh_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def editing_remote_samplesheet(study):
+            logging.info(f"Editing remote samplesheet for study {study['study']}...")
+            command = f"sed -i 's|{BASE_PATH}/|{REMOTE_PATH}/|g' {REMOTE_PATH}/{study['study']}/samplesheet.csv"
+            ssh_op = CustomSSHOperator(
+                task_id='editing_remote_samplesheet_' + study['study'],
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SSH,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                command=command
+            )
+            ssh_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def submit_nextflow_script(script_info):
+            logging.info(f"Submitting Nextflow script for study {script_info['study']['study']}...")
+            study = script_info["study"]
+            script_name = script_info["script_name"]
+            command = f'bsub -n 64 -M 250000 -R "rusage[mem=250000] span[hosts=1]" sh {REMOTE_PATH}/{study["study"]}/{script_name}'
```

Comment:
the parent task is almost guaranteed to not require 250gb. and the subtasks will request for themselves what resources they think they need. if the subtasks are wrong about what resources they need, then the place to change those resource requirements will be the ampliseq.config file.